evergreen.yml

```diff
@@ -104,17 +132,92 @@ functions:
       type: test
       params:
         shell: bash
-        working_dir: mongo-odbc-driver
+        working_dir: mongosql-odbc-driver
         script: |
           ${prepare_shell}
           cargo build --release
 
+  "upload artifact":
+    - command: s3.put
+      params:
+        build_variants:
+          - windows-32
+          - windows-64
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        local_file: mongosql-odbc-driver/target/release/mongoodbc.dll
+        remote_file: mongosql-odbc-driver/artifacts/test/${version_id}/${build_variant}/mongoodbc.dll
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/octet-stream
+
+  "download artifact":
+    - command: s3.get
+      params:
+        build_variants:
+          - windows-32
+          - windows-64
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        remote_file: mongosql-odbc-driver/artifacts/test/${version_id}/${build_variant}/mongoodbc.dll
+        local_file: mongosql-odbc-driver/mongoodbc.dll
+        bucket: mciuploads
+
+  "setup driver on Windows":
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: mongosql-odbc-driver
+        script:  |
+          ${prepare_shell}
+          sed -i 's@%DRIVER_DLL_PATH%@'"$DRIVER_DLL_PATH"'@' setup/setupDSN.reg
+          if [ 'windows-32' == '${build_variant}' ]; then
+            sed -i 's@SOFTWARE@'"SOFTWARE\\\\Wow6432Node"'@' setup/setupDSN.reg
+          fi
+          reg import "setup\setupDSN.reg"
+          echo "----- Registry entries after setup ----"
+          reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\ODBC\ODBCINST.INI\ODBC Drivers" -s
+          reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ODBC Drivers" -s
+          if [ 'windows-32' == '${build_variant}' ]; then
+            reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER" -s
+            reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER" -s
+          else
+            reg query "HKEY_LOCAL_MACHINE\SOFTWARE\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER" -s
+            reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\ODBC\ODBCINST.INI\ADL_ODBC_DRIVER" -s
+          fi
+          echo "-------------------------"
+        continue_on_err: true
```

Comment:
What are the situations where we'd expect this to fail but would still want to continue? I'd have expected that if driver setup failed, we'd want the task to fail

Reply:
This was because if a registry entry doesn't exist, the task fails. But a couple of registry entries are expected to be missing. The idea is to make sure the setup did not create unwanted entries and/or that they are no lingering entries. I changed this to suppress errors when we expect the entry to be missing.